### DISCUSSION TITLE
Curious, how many BORIS users are able to copy/paste events?

### DISCUSSION THREAD

**Discussion by SirWilksBoothDaThird** [olivierfriard/BORIS#798] [General]
I've enquired multiple times on this issue in the BORIS tech. issues threads since it seems to be one shrouded in mystery and hasn't been brought up before. I'm interested in other users' experience with the copy and paste functions for events. 

Are people able to copy and paste events within the events table in BORIS 8.x versions, or am I the only one using the newest version of BORIS (full system install) and the newest Python library installations who is able to successfully copy an event but is unable to paste anything?

Pasting elsewhere works, and I've gotten events to paste seemingly by accident maybe twice out of a couple of hundred tries.
I know that the feature is discontinued in BORIS 7.x (as is the entire version), but I am puzzled as to why it still exists in the program in the newest version 8.27.10, works "half-way", and is continually present in the online BORIS documentation.

Please, everyone who is using any of the latest versions, let me know your experiences because this might shed some light on the issue for me.

Thanks in advance to all as well as the developers of this very useful software!

**Comment by paulagsotres** [comment]
Hello!

In the version 9.6.5, you can copy and paste events with no problem! What it's a bit tricky is the format to paste the events. 

The format should be:
[Time in seconds]	[Subject]	[Code]	[Modifier]	[Comment]	[Frame index]

For example:
13.500	1	close_to_roi1	None	Test	270

Each of the columns should be separated with a Tab (\t) and you can leave empty spaces. I have made a small simple code in Python that might help you transform your exported tabular events (exported as excel files from Observation > Export Observations > Tabular events) into a _pastable_ format. 


-------
import pandas as pd
import os
excel_path = r"C:\Users\pgomez\Desktop\mouse.xlsx"  # replace with your actual file path
df = pd.read_excel(excel_path)
df_formatted = df[['Time', 'Subject', 'Behavior', 'Image file path', 'Comment', 'Image index']]  # adjust if needed
df_formatted.columns = ['Time', 'Subject', 'Code', 'Modifier', 'Comment', 'Frame index']
base_name = os.path.splitext(excel_path)[0]  # removes .xlsx
txt_path = f"{base_name}.txt"
df_formatted.to_csv(txt_path, sep='\t', index=False)
print(f"TXT file created: {txt_path}")


--------

If you copy/paste the content of the resulting text file in the panel of observations, the events should be imported. You don't need to copy the column name row. Let me know if that works for you. 

Best, 
Paula
